processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -446,41 +446,48 @@ private void mergeInheritedOptions(SourceMethod method, MapperConfiguration mapp
                 initializingMethods,
                 mapperConfig );
 
+        MappingInheritanceStrategyPrism inheritanceStrategy = mapperConfig.getMappingInheritanceStrategy();
+
         if ( templateMappingOptions != null ) {
             mappingOptions.applyInheritedOptions( templateMappingOptions, false, method, messager, typeFactory );
         }
         else if ( inverseMappingOptions != null ) {
             mappingOptions.applyInheritedOptions( inverseMappingOptions, true, method, messager, typeFactory );
         }
-        else if ( mapperConfig.getMappingInheritanceStrategy() == AUTO_INHERIT_FROM_CONFIG ) {
-            if ( applicablePrototypeMethods.size() == 1 ) {
-                mappingOptions.applyInheritedOptions(
-                    first( applicablePrototypeMethods ).getMappingOptions(),
-                    false,
-                    method,
-                    messager,
-                    typeFactory );
-            }
-            else if ( applicablePrototypeMethods.size() > 1 ) {
-                messager.printMessage(
-                    method.getExecutable(),
-                    Message.INHERITCONFIGURATION_MULTIPLE_PROTOTYPE_METHODS_MATCH,
-                    Strings.join( applicablePrototypeMethods, ", " ) );
+        else if ( inheritanceStrategy.isAutoInherit() ) {
+
+            if ( inheritanceStrategy.isApplyForward() ) {
+                if ( applicablePrototypeMethods.size() == 1 ) {
+                    mappingOptions.applyInheritedOptions(
+                        first( applicablePrototypeMethods ).getMappingOptions(),
+                        false,
+                        method,
+                        messager,
+                        typeFactory );
+                }
+                else if ( applicablePrototypeMethods.size() > 1 ) {
+                    messager.printMessage(
+                        method.getExecutable(),
+                        Message.INHERITCONFIGURATION_MULTIPLE_PROTOTYPE_METHODS_MATCH,
+                        Strings.join( applicablePrototypeMethods, ", " ) );
+                }
             }
 
-            if ( applicableReversePrototypeMethods.size() == 1 ) {
-                mappingOptions.applyInheritedOptions(
-                    first( applicableReversePrototypeMethods ).getMappingOptions(),
-                    true,
-                    method,
-                    messager,
-                    typeFactory );
-            }
-            else if ( applicableReversePrototypeMethods.size() > 1 ) {
-                messager.printMessage(
-                    method.getExecutable(),
-                    Message.INHERITINVERSECONFIGURATION_MULTIPLE_PROTOTYPE_METHODS_MATCH,
-                    Strings.join( applicablePrototypeMethods, ", " ) );
```

Comment:
I just realized that this was wrong all along 😄. It seems we don't have a test for this error case